src/main/java/de/hysky/skyblocker/mixins/HandledScreenMixin.java

```diff
@@ -215,8 +216,10 @@ protected HandledScreenMixin(Text title) {
             }
         }
 
-        if (currentSolver != null) {
-            SkyblockerMod.getInstance().containerSolverManager.onSlotClick(slotId, stack);
+        if (currentSolver instanceof TerminalSolver) {
```

Comment:
This makes it so that `onSlotClick` only gets called on `TerminalSolver`s, the original null check is probably better in case we add a non-terminal solver that depends on this.